### PR TITLE
AR-2148 feedback changes wallet SDK

### PR DESCRIPTION
**Pull request by karthik-durai**
1. Close icon on the header should be white in colour in dark mode and black in light mode.
2. Border radius on the wallet frame needs to be updated per the design.
3. Add request numbers badge to bubble and relevant logic to display pending request count.
4. Add close button and relevant logic to destroy wallet.

screenshots:

Screenshot 2022-04-14 at 4 43 51 PM

Screenshot 2022-04-14 at 4 46 09 PM

Screenshot 2022-04-14 at 4 46 52 PM

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -1,113 +1,126 @@
...
+      },
+    })
+    this.arcanaProvider.setConnection(communication)
+    this.arcanaProvider.setHandlers(
+      this.iframeWrapper.show,
+      this.iframeWrapper.hide
+    )
+  }
+
+  destroyWalletUI() {
```
Comment (makylfang) [inline src/index.ts:77]: small nitpick but you can make it like
```
destroyWalletUI = () => {} 
```
so you wont have to bind it to `this`

**Review by makylfang — APPROVED** [review]
LGTM.

File: src/iframeWrapper.ts [hunk 2]
```diff
@@ -151,13 +152,27 @@ export default class IframeWrapper {
...
         src: getLogo(this.themeConfig, "vertical"),
       })
+      const reqCountBadge = createDomElement("p", {
+        id: "req-count-badge",
+        style: widgetBubbleStyle.reqCountBadge,
+        textContent: 0,
+      })
+      const closeButton = createDomElement("button", {
+        onclick: (event: Event) => {
+          event.stopImmediatePropagation()
```
Comment (ajithranka) [inline src/iframeWrapper.ts:162]: Why are we using `stopImmediatePropagation` instead of `stopPropagation`?

File: src/iframeWrapper.ts [hunk 3]
```diff
@@ -29,7 +29,8 @@ export default class IframeWrapper {
   constructor(
     private params: iframeWrapperParams,
     private iframeUrl: string,
-    private themeConfig: IWidgetThemeConfig
+    private themeConfig: IWidgetThemeConfig,
+    private destroyWalletUI: Function
```
Comment (ajithranka) [inline src/iframeWrapper.ts:33]: Using `Function` doesn't help with type-checking the function parameters or return value. Can we annotate them explicitly?

```suggestion
    private destroyWalletUI: (this: WalletProvider) => void
```

File: src/iframeWrapper.ts [hunk 4]
```diff
@@ -151,13 +152,27 @@ export default class IframeWrapper {
       const buttonLogo = createDomElement("img", {
         src: getLogo(this.themeConfig, "vertical"),
       })
+      const reqCountBadge = createDomElement("p", {
+        id: "req-count-badge",
+        style: widgetBubbleStyle.reqCountBadge,
```
Comment (ajithranka) [inline src/iframeWrapper.ts:157]: We don't want to show the request count badge when there are no active requests. Can we add a class to toggle this when textCount is 0?

**Review by ajithranka — COMMENTED** [review]
Just a note: The PR is a bit difficult to follow since it's mixing new features with linting fixes. Please separate these out into different PRs next time.

File: src/iframeWrapper.ts (hunk 4 above)
Comment (karthik-durai) [reply src/iframeWrapper.ts:157]: okay, working on it.

File: src/iframeWrapper.ts (hunk 3 above)
Comment (karthik-durai) [reply src/iframeWrapper.ts:33]: working on it

**Comment by karthik-durai** [comment]
Creating New PR without **code format** changes.

**Pull request closed without merging**